File: mazes-for-programmers/python/src/mazes/polar_cell.py
from cell import Cell


class PolarCell(Cell):
    def __init__(self, row, column):
        super().__init__(row, column)
        self.cw = None
        self.ccw = None
        self.inward = None
        self.outward = []

    def neighbors(self):
        result = []
        if self.cw is not None:
            result.append(self.cw)
        if self.ccw is not None:
            result.append(self.ccw)
        if self.inward is not None:
            result.append(self.inward)
        result.extend(self.outward)
        return result

File: mazes-for-programmers/kotlin/src/main/kotlin/io/github/ocirne/mazes/PolarCell.kt
package io.github.ocirne.mazes

class PolarCell(override val row: Int, override val column: Int) : Cell(row, column) {

    var cw: Cell? = null
    var ccw: Cell? = null
    var inward: Cell? = null
    val outward: MutableList<Cell> = mutableListOf()

    override fun neighbors(): List<Cell?> {
        return arrayListOf(cw, ccw, inward).filterNotNull() + outward
    }
}
